clippy_lints/src/operators/arithmetic_side_effects.rs

```diff
@@ -129,21 +123,53 @@ impl ArithmeticSideEffects {
             self.issue_lint(cx, expr);
         }
     }
+
+    fn manage_unary_ops<'tcx>(
+        &mut self,
+        cx: &LateContext<'tcx>,
+        expr: &hir::Expr<'tcx>,
+        un_expr: &hir::Expr<'tcx>,
+        un_op: hir::UnOp,
+    ) {
+        if constant_simple(cx, cx.typeck_results(), un_expr).is_some() {
```

Comment:
Would it be better to ignore all constants here? At least for the builtin types. I don't think a constant changing between the minimum value and another is particularly common.

Reply:
Using `constant` instead of `constant_simple`? I tried such approach without success because `constant` filters out `let _n = -i32::MIN;` in a runtime context. On the constant side, `const _N: i32 = -i32::MIN;` won't even let Clippy kick in.